src/index.ts

```diff
@@ -64,26 +40,34 @@ function generateVerifier(length: number): string {
  * @param code_verifier
  * @returns The base64 url encoded code challenge
  */
-export function generateChallenge(code_verifier: string) {
-  return SHA256(code_verifier).toString(encBase64url);
+export async function generateChallenge(code_verifier: string) {
+  const buffer = await crypto.subtle.digest(
+    "SHA-256",
+    new TextEncoder().encode(code_verifier)
+  );
+  // Generate base64url string
+  return btoa(String.fromCharCode(...new Uint8Array(buffer)))
```

Comment:
I am getting a warning that btoa is deprecated
> The signature '(data: string): string' of 'btoa' is deprecated.ts(6387)

The jsdoc (in here node_modules/@types/node/buffer.d.ts) suggest using `buf.toString('base64')` instead.

Reply:
For Node.js context yes it's deprecated, but there's no good web standard to replace this. This exists in Node.js to be compatible with web environment and thus IMO it should be fine here for this specific purpose.

Reply:
```suggestion
  // btoa is deprecated in Node.js but is used here for web browser compatibility
  // (which has no good replacement yet, see also https://github.com/lucacasonato/proposal-binary-encoding)
  return btoa(String.fromCharCode(...new Uint8Array(buffer)))
```

Reply:
Okay, I understand the concern is more to do with strings representing binary data. In this case since we know what the strings are ahead of time, it shouldn't be an issue. Thanks for adding clarifying comments.